pallets/api/src/fungibles/mod.rs

```diff
@@ -0,0 +1,122 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::{DispatchResult, DispatchResultWithPostInfo, WithPostDispatchInfo},
+		pallet_prelude::*,
+		traits::fungibles::{
+			approvals::Inspect as ApprovalInspect, metadata::Inspect as MetadataInspect, Inspect,
+		},
+	};
+	use frame_system::pallet_prelude::*;
+	use pallet_assets::WeightInfo;
+	use sp_runtime::traits::StaticLookup;
+
+	use primitives::constants::fungibles::*;
+
+	type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+	type AssetIdOf<T> = <pallet_assets::Pallet<T, AssetsInstanceOf<T>> as Inspect<
+		<T as frame_system::Config>::AccountId,
+	>>::AssetId;
+	type AssetIdParameterOf<T> =
+		<T as pallet_assets::Config<AssetsInstanceOf<T>>>::AssetIdParameter;
+	type Assets<T> = pallet_assets::Pallet<T, AssetsInstanceOf<T>>;
+	type AssetsInstanceOf<T> = <T as Config>::AssetsInstance;
+	type AssetsWeightInfo<T> = <T as pallet_assets::Config<AssetsInstanceOf<T>>>::WeightInfo;
+	type BalanceOf<T> = <pallet_assets::Pallet<T, AssetsInstanceOf<T>> as Inspect<
+		<T as frame_system::Config>::AccountId,
+	>>::Balance;
+
+	/// The required input for state queries in pallet assets.
+	#[derive(Encode, Decode, Debug, MaxEncodedLen)]
+	#[repr(u8)]
+	#[allow(clippy::unnecessary_cast)]
+	pub enum Keys<T: Config> {
```

Comment:
Not sure Keys is the correct name, as those values arent explicitly keys. Also enum naming convention is typically singular, as one typically only specifies one value at a type. Different if bitflags mind.

Reply:
Changed it to `FungibleKey`

Reply:
Technically speaking one of these implementations could look up multiple values, combining them into or calculating a result. Something more abstract like 'State' or 'Query' or 'StateQuery' might be a better starting name, mostly as people just tend to copy some existing pattern.

This may suggest an improvement to the corresponding extension name too.